### PR TITLE
Generate extensions for protocolsMap.py

### PR DESCRIPTION
**Pull request by kimdv**
Generate extensions for [protocolsMap.py](https://github.com/apple/swift/blob/main/utils/gyb_syntax_support/protocolsMap.py).

### CODE REVIEW

File: Sources/SwiftSyntaxBuilder/Buildables.swift.gyb [hunk 1]
```diff
@@ -264,57 +265,17 @@ extension TokenSyntax: ExpressibleAsTokenSyntax {
...
-% for protocol, conformances in SYNTAX_BUILDABLE_EXPRESSIBLE_AS_CONFORMANCES.items():
-%   if 'ExpressibleAsConditionElementList' in conformances:
-extension ${protocol} {
-  public func createConditionElementList() -> ConditionElementList {
-    ConditionElementList([self])
-  }
-}
-
-%   end
-% end
```
Comment (kimdv) [inline Sources/SwiftSyntaxBuilder/Buildables.swift.gyb:277]: This one properly slipped through when added #334

File: Sources/SwiftSyntaxBuilder/Buildables.swift.gyb [hunk 2]
```diff
@@ -264,57 +265,17 @@ extension TokenSyntax: ExpressibleAsTokenSyntax {
...
-
 % for protocol, conformances in SYNTAX_BUILDABLE_EXPRESSIBLE_AS_CONFORMANCES.items():
-%   if 'ConditionElement' in conformances:
+%   for conformance in conformances:
+%     node = NODE_MAP.get(conformance)
+%     if node and node.children:
+%       param = filter(lambda x : x.type_name != "TokenSyntax", node.children)[0].swift_name
```
Comment (ahoppen) [inline Sources/SwiftSyntaxBuilder/Buildables.swift.gyb:272]: I don’t think this is correct. We want to the first child that doesn’t have a default value. Currently optional children also have a default value of `nil` and `TokenSyntax` children might not have a default value if they don’t have a default text (see https://github.com/apple/swift/blob/17f1cf92973d2c1b3d5fe6b70f4a33771a303e3a/utils/gyb_syntax_support/kinds.py#L58-L64)

I think a better implementation would be 

```python
non_defaulted_params = filter(lambda child : syntax_buildable_default_init_value(child, SYNTAX_TOKEN_MAP.get(child.syntax_kind)) == None, node.children)
assert len(non_defaulted_params) == 1, "ExpressibleAs conformances expects the conforming type to have an initializer with a single non-optional child"
param = non_defaulted_params[0].swift_name
```

Comment (kimdv) [reply Sources/SwiftSyntaxBuilder/Buildables.swift.gyb]: Ah, I didn't see that. 
Added. Tough changes the `== None` to `== ""` as we return an empty string.

**Comment by kimdv** [comment]
@swift-ci Please test

**Review by ahoppen — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Pull request merged**